pkg/api/v2/types.go

```diff
@@ -777,6 +801,7 @@ type ExternalClusterMachineDeployment struct {
 type ExternalClusterMDPhase struct {
```

Comment:
should we rename this ExternalClusterMDPhase => ExternalClusterMDStatus. wdyt? 
If yes, then we can rename 
AKSMDPhase  => AKSMDStatus